worker/upgradesteps/worker.go

```diff
@@ -59,7 +59,7 @@ var (
 func NewLock(agentConfig agent.Config) gate.Lock {
 	lock := gate.NewLock()
 
-	if wrench.IsActive("machine-agent", "always-try-upgrade") {
+	if wrench.IsActive("machine-agent", "always-try-upgrade") || wrench.IsActive("unit-agent", "always-try-upgrade") {
```

Comment:
this will check for machine-agent in a unit agent, and unit-agent in a machine agent
how about this instead?

```go
wrenchKey := agentConfig.Tag().Kind() + "-agent" // (machine|unit)-agent
if wrench.IsActive(wrenchKey, "always-try-upgrade") {
```

Reply:
good idea \o/ done :)